### DISCUSSION TITLE
apollo客户端使用自定义的logback-spring.xml文件并设置热更新

### DISCUSSION THREAD

**Discussion by hhh220** [apolloconfig/apollo#4069] [Q&A]
apollo客户端怎样使用自定义的logback-spring.xml文件并设置热更新？

**Comment by hhh220** [comment]
apollo的application配置了spring.application.name =xxx，项目本地application配置了spring.application.name =xxx2。logback在打印日志的时候取的是xxx2，apollo生成的本地缓存文件目录中取的xxx。这两者在某种情况下会导致异常吗？
logback配置截图：
![image](https://user-images.githubusercontent.com/92715270/139041933-e7b23d6a-d051-49af-afc1-10da1ee10f2c.png)

**Reply by hhh220** [reply]
将logback-spring.xml中source="spring.application.name"更改为source="app.id"解决

**Reply by hhh220** [reply]
apollo配置项中指定的“logging.level.com.pcitc.pay = warn”无法覆盖本地logback-spring.xml中的日志级别INFO
![image](https://user-images.githubusercontent.com/92715270/139222698-cce6e6b3-db21-4487-bf3e-bc83e071f1e8.png)

**Reply by nobodyiam** [reply]
You may reference [spring-boot-logger](https://github.com/apolloconfig/apollo-use-cases/tree/master/spring-boot-logger) for more information.

**Reply by hhh220** [reply]
3q，搞定了。下一步就是测试多环境部署了
